gulpfile.js

```diff
@@ -0,0 +1,381 @@
+    '@angular', 'core-js/client', 'hammerjs', 'rxjs', 'systemjs/dist', 'zone.js/dist'
+  ];
+
+  return gulpMerge(
+    npmVendorFiles.map(function(root) {
+      const glob = path.join(root, '**/*.+(js|js.map)');
+      return gulp.src(path.join('node_modules', glob))
+        .pipe(gulp.dest(path.join('dist/vendor', root)));
+    }));
+});
+
+
+/***************************************************************************************************
+ * DevApp Build Tasks.
+ */
+gulp.task(':build:devapp:ts', [':build:components:ts'], createTsBuildTask({ tsConfigPath: devAppDir }));
+gulp.task(':build:devapp:scss', [':build:components:scss'], createSassBuildTask({
+  dest: outDir,
+  root: devAppDir,
+  // Change this once we have a better strategy for releasing SCSS files.
+  includePaths: [
+    path.join(componentsDir, 'core/style'),
+    componentsDir
+  ]
+}));
+gulp.task(':build:devapp:assets', function() {
+  return gulp.src(path.join(devAppDir, '**/*'))
+    .pipe(gulp.dest(outDir));
+});
+
+gulp.task('build:devapp', [
+  'build:components',
+  ':build:vendor',
+  ':build:devapp:ts',
+  ':build:devapp:scss',
+  ':build:devapp:assets'
+]);
+
+/***************************************************************************************************
+ * DevApp Build Tasks.
+ */
+gulp.task(':build:e2eapp:ts', [':build:components:ts'], createTsBuildTask({ tsConfigPath: e2eAppDir }));
+gulp.task(':build:e2eapp:scss', [':build:components:scss'], createSassBuildTask({
+  dest: outDir,
+  root: e2eAppDir,
+  // Change this once we have a better strategy for releasing SCSS files.
+  includePaths: [
+    path.join(componentsDir, 'core/style'),
+    componentsDir
+  ]
+}));
+gulp.task(':build:e2eapp:assets', function() {
+  return gulp.src(path.join(e2eAppDir, '**/*'))
+    .pipe(gulp.dest(outDir));
+});
+
+gulp.task('build:e2eapp', [
+  'build:components',
+  ':build:vendor',
+  ':build:e2eapp:ts',
+  ':build:e2eapp:scss',
+  ':build:e2eapp:assets'
+]);
+
+
+/***************************************************************************************************
+ * Global tasks.
+ */
+gulp.task('default', ['help']);
+
+gulp.task('help', function() {
+  const tasks = Object.keys(gulp.tasks)
+    .filter(x => !x.startsWith(':'))
+    .filter(x => !x.startsWith('ci:'))
+    .filter(x => x != 'default')
+    .sort();
+
+  console.log(`\nHere's a list of supported tasks:\n   `, tasks.join('\n    '));
+  console.log(`\nYou're probably looking for "test" or "serve:devapp".\n\n`);
+});
+
+gulp.task('build', ['build:devapp']);
+
+gulp.task('clean', function() {
+  return gulp.src('dist', { read: false })
+    .pipe(gulpClean());
+});
+gulp.task(':clean:spec', function() {
+  return gulp.src('dist/**/*.spec.*', { read: false })
+    .pipe(gulpClean());
+});
+gulp.task(':clean:assets', function() {
+  return gulp.src('dist/**/*+(.html|.css)', { read: false })
+    .pipe(gulpClean());
+});
+
+gulp.task('lint', createExecTask('tslint', ['-c', 'tslint.json', 'src/**/*.ts']));
```

Comment:
Let's call this `tslint` and then have a `gulp lint` that runs tslint, stylelint, and madge.